src/workload.cpp

```diff
@@ -673,197 +722,60 @@ rhs_t Comp::get_rhs() const {
-            os << get<Decr>(wl_spec);
-            break;
-        }
-        // Uniq
-        case 4: {
-            os << get<Unique>(wl_spec);
-            break;
-        }
-        default: break;
-    }
+ostream& operator<<(ostream& os, const WlSpec* wl_spec) {
+    os << wl_spec->to_string();
     return os;
 }
 
-bool operator==(const wl_spec_t& spec1, const wl_spec_t& spec2) {
-    if (holds_alternative<Comp>(spec1) && holds_alternative<Comp>(spec2)) {
-        return get<Comp>(spec1) == get<Comp>(spec2);
-    }
-    if (holds_alternative<Same>(spec1) && holds_alternative<Same>(spec2)) {
-        return get<Same>(spec1) == get<Same>(spec2);
-    }
-    if (holds_alternative<Incr>(spec1) && holds_alternative<Incr>(spec2)) {
-        return get<Incr>(spec1) == get<Incr>(spec2);
-    }
-    if (holds_alternative<Decr>(spec1) && holds_alternative<Decr>(spec2)) {
-        return get<Decr>(spec1) == get<Decr>(spec2);
-    }
-    return false;
-}
-
-bool operator<(const wl_spec_t& spec1, const wl_spec_t& spec2) {
-    // Comp < Same < Incr < Decr < Uniq
-
-    if (holds_alternative<Comp>(spec1)) {
-        if (holds_alternative<Comp>(spec2)) {
-            return get<Comp>(spec1) < get<Comp>(spec2);
-        } else {
-            return true;
-        }
-    }
-
-    else if (holds_alternative<Same>(spec1)) {
-        if (holds_alternative<Same>(spec2)) {
-            return get<Same>(spec1) < get<Same>(spec2);
-        } else if (holds_alternative<Comp>(spec2)) {
-            return false;
-        } else {
-            return true;
-        }
-    }
-
-    else if (holds_alternative<Incr>(spec1)) {
-        if (holds_alternative<Incr>(spec2)) {
-            return get<Incr>(spec1) < get<Incr>(spec2);
-        } else if (holds_alternative<Comp>(spec2) || holds_alternative<Same>(spec2)) {
-            return false;
-        } else {
-            return true;
-        }
-    }
-
-    else if (holds_alternative<Decr>(spec1)) {
-        if (holds_alternative<Decr>(spec2)) {
-            return get<Decr>(spec1) < get<Decr>(spec2);
-        } else {
-            return false;
-        }
-    }
-
-    else if (holds_alternative<Unique>(spec1)) {
-        if (holds_alternative<Unique>(spec2)) {
-            return get<Unique>(spec1) < get<Unique>(spec2);
-        } else if (holds_alternative<Unique>(spec2)) {
-            return false;
-        } else {
-            return true;
-        }
-    }
-    cout << "operator < for wl_spec_t: should not reach here" << endl;
-    return false;
-}
-
 //************************************* TimedSpec *************************************//
 
 // Invariant: An object of this class
 //            is always normalized independent of the operation
 //            running on it.
 
-TimedSpec::TimedSpec(wl_spec_t wl_spec, time_range_t time_range, unsigned int total_time):
-wl_spec(wl_spec),
+TimedSpec::TimedSpec(WlSpec* wl_spec, time_range_t time_range, unsigned int total_time):
+wl_spec(std::move(wl_spec)),
```

Comment:
Why move used here?

Reply:
Used to be a smart pointer, which supports move semantics. Forgot to remove it when we changed back to raw pointers.

[Remove unnecessary std::move](https://github.com/all-things-networking/fperf/pull/27/commits/cc3af7ab684acdeea1eb0b1e8b3d598c34b97541)